src/deadline/unreal_adaptor/UnrealClient/step_handlers/unreal_render_step_handler.py

```diff
@@ -107,52 +109,111 @@ def on_begin_frame(self):
+        :param error: The error message
+        """
         logger.error(f"Render Executor: Error: {error}")
 
     @staticmethod
-    def executor_finished_callback(movie_pipeline=None, results=None):
+    def executor_finished_callback(pipeline_executor=None, success=None):
+        """
+        Callback executed when RemoteRenderMoviePipelineEditorExecutor finished render
+
+        :param pipeline_executor: The RemoteRenderMoviePipelineEditorExecutor instance
+        :param success: Whether finished successfully or not
+        """
         logger.info("Render Executor: Rendering is complete")
 
     @staticmethod
-    def create_queue_from_manifest(movie_pipeline_queue_subsystem, queue_manifest_path: str):
+    def copy_pipeline_queue_from_manifest_file(
+        movie_pipeline_queue_subsystem, queue_manifest_path: str
+    ):
         """
-        Create the unreal.MoviePipelineQueue object from the given queue manifest path
+        Create unreal.MoviePipelineQueue from manifest file by loading the file.
+        Unreal requires the manifest file to be placed under the <project_root>/Saved directory.
 
-        :param movie_pipeline_queue_subsystem: The unreal.MoviePipelineQueueSubsystem instance
+        :param movie_pipeline_queue_subsystem: unreal.MoviePipelineQueueSubsystem instance
         :param queue_manifest_path: Path to the manifest file
         """
-        queue_manifest_path = queue_manifest_path.replace("\\", "/")
         manifest_queue = unreal.MoviePipelineLibrary.load_manifest_file_from_string(
             queue_manifest_path
         )
-
         pipeline_queue = movie_pipeline_queue_subsystem.get_queue()
         pipeline_queue.delete_all_jobs()
         pipeline_queue.copy_from(manifest_queue)
 
+    @staticmethod
+    def create_queue_from_manifest(movie_pipeline_queue_subsystem, queue_manifest_path: str):
+        """
+        Create the unreal.MoviePipelineQueue object from the given queue manifest path.
+
+        Copy manifest file to `<project_root>/Saved directory` if manifest located outside
+        the project folder (case when project synced via P4/UGS to workspaces root and manifest file
+        passed via S3 and placed in OpenJob `<session_directory>/<asset_root>`)
+
+        :param movie_pipeline_queue_subsystem: The unreal.MoviePipelineQueueSubsystem instance
+        :param queue_manifest_path: Path to the manifest file
+        """
+        manifest_path = queue_manifest_path.replace("\\", "/")
+        project_saved_dir = unreal.SystemLibrary.get_project_saved_directory().rstrip("/")
+
+        if not manifest_path.startswith(project_saved_dir):
+            project_manifest_directory = os.path.join(
+                project_saved_dir, "UnrealDeadlineCloudService", "RenderJobManifests"
+            ).replace("\\", "/")
+            os.makedirs(project_manifest_directory, exist_ok=True)
+            saved_manifest_file_path = shutil.copy(manifest_path, project_manifest_directory)
```

Comment:
Testing out these changes (Without perforce or ugs) I ran into an issue here - If I have multiple tasks (multiple shots to render), I'm falling into this section each time because my manifest_path is something like:

C:/ProgramData/Amazon/OpenJD/session-881e54a12aff4418a0302d30025f1022lmyoqih9/assetroot-0a339ba0c0ce03618ff1/AppData/Local/UnrealEngine/5.4/Saved/UnrealDeadlineCloudService/RenderJobManifests/RenderJobManifest3EC2BB6643D3259C800C5BA086925022.utxt

And my project_saved_dir is:

C:/ProgramData/Amazon/OpenJD/session-881e54a12aff4418a0302d30025f1022lmyoqih9/assetroot-0a339ba0c0ce03618ff1/Documents/Unreal Projects/MeerkatDemo7/Saved

which fails the startswith check.  This doesn't seem correct based on the comment above since I'm not using P4 or UGS, but the bigger error is that on the second time through the file already exists at the destination and something has it open so I'm getting a Permission denied error and never succeeding on my second task.  Even worse it's also then not counting as a failure, it just keeps spinning on task 2.

So the first/easiest fix seems to be to check if that file exists, I can get things passing and the job succeeding if I change this section to something like:

`            
            os.makedirs(project_manifest_directory, exist_ok=True)
            output_path = os.path.join(project_manifest_directory, os.path.basename(manifest_path))
            if not os.path.exists(output_path):
                saved_manifest_file_path = shutil.copy(manifest_path, output_path) 
                manifest_path = saved_manifest_file_path
            else:
                manifest_path = output_path`

But it seems like those other issues are probably also worth looking into

Reply:
Thats really strange and makes me ask even more questions to clarify. So, from the paths I see that your project located in some `C:/Documents/Unreal Projects/MeerkatDemo7`

> And my project_saved_dir is:
> 
> C:/ProgramData/Amazon/OpenJD/session-881e54a12aff4418a0302d30025f1022lmyoqih9/assetroot-0a339ba0c0ce03618ff1/Documents/Unreal Projects/MeerkatDemo7/Saved

But Unreal submitter save manifest to `%LOCALAPPDATA%/UnrealEngine/5.4/Saved/UnrealDeadlineCloudService/RenderJobManifests/RenderJobManifest3EC2BB6643D3259C800C5BA086925022.utxt`

> my manifest_path is something like:
> 
> C:/ProgramData/Amazon/OpenJD/session-881e54a12aff4418a0302d30025f1022lmyoqih9/assetroot-0a339ba0c0ce03618ff1/AppData/Local/UnrealEngine/5.4/Saved/UnrealDeadlineCloudService/RenderJobManifests/RenderJobManifest3EC2BB6643D3259C800C5BA086925022.utxt

Right?

Seems that `unreal.SystemLibrary.get_project_saved_directory` on your machine point to %LOCALAPPDATA%  which is even stranger and wrong, because at least the path should contain the project name. Will investigate it and maybe change locating the saved directory to something different